src/Player/playerMovementUtils.ts

```
@@ -1,20 +1,20 @@
-import { PlayerPosition, PlayerVelocity } from '../allTypes';
+import { PlayerPosition, PlayerVelocity } from '../types';
 /**
  * serialize is a helper function that is a light wrapper around stringifying a JSON object.
- * @param data 
+ * @param data
  * @returns string
  */
-export function serialize(data:PlayerPosition):string {
+export function serialize(data: PlayerPosition): string {
   return JSON.stringify(data);
 }
 
 /**
  * deserialize is a helper function that is a light wrapper around parsing a JSON string.
  * Returns undefined if error occured.
- * @param data 
+ * @param data
  * @returns string
  */
-export function deserialize(data:PlayerPosition) {
+export function deserialize(data: PlayerPosition) {
```

If we're going to try-catch here let's report a meaningful error?

---

Agree! Returning undefined here is not as descriptive or clear as it could be

---

I still think the best way to handle this is to either
a) not handle the error, which will throw an exception and make the program crash (not recommended in this case -- soft errors should not cause crashes)
b) properly handle the error, which should do some logic to make sure the program does not crash and return the data to the user.

Optional types should, in general, reveal meaningful information, not cover exceptions. Exceptions are thrown for a reason, and it can be dangerous to let them go undetected and handle them as `null`. I think that if there exception handling to be done, it should be done at this level (i.e. don't make an upstream function handle the `undefined`, maybe return an empty JSON object instead?) The upstream function shouldn't have to deal with something failing further down.